pages/accountLists/[accountListId].page.test.tsx

```diff
@@ -108,4 +108,91 @@ describe('AccountListsId page', () => {
       expect(getByText('Good Morning, firstName.')).toBeInTheDocument();
     });
   });
+  describe('AccountListId Redirects', () => {
+    const restApiNotFoundErrorMessage = "Resource 'AccountList' is not valid";
+
+    it('replaces and redirects to the default account list id if Not Found', async () => {
+      const query = jest.fn();
+
+      (makeSsrClient as jest.Mock).mockReturnValue({
+        query: query,
+      });
+      query.mockRejectedValueOnce(new Error(restApiNotFoundErrorMessage));
+
+      query.mockResolvedValueOnce({
+        data: {
+          user: {
+            defaultAccountList: 'default-id',
+          },
+        },
+      });
+
+      const { redirect } = (await getServerSideProps({
+        req: { url: '/accountLists/[accountListId]/contacts' },
+        query: {
+          accountListId: 'account-list-1',
+        },
+      } as unknown as GetServerSidePropsContext)) as GetServerSidePropsReturn;
+
+      expect(redirect).toEqual({
+        destination: '/accountLists/default-id/contacts',
+        permanent: false,
+      });
+    });
+
+    it('redirects to the account list selector page if theres no defaultAccountList', async () => {
+      const query = jest.fn();
+
+      (makeSsrClient as jest.Mock).mockReturnValue({
+        query: query,
+      });
+      query.mockRejectedValueOnce(new Error(restApiNotFoundErrorMessage));
+
+      query.mockResolvedValueOnce({
+        data: {
+          user: {
+            defaultAccountList: null,
+          },
+        },
+      });
+
+      const { redirect } = (await getServerSideProps({
+        req: { url: '/accountLists/[accountListId]/contacts' },
+        query: {
+          accountListId: 'account-list-1',
+        },
+      } as unknown as GetServerSidePropsContext)) as GetServerSidePropsReturn;
+
+      expect(redirect).toEqual({
+        destination: '/accountLists',
+        permanent: false,
+      });
```

Comment:
I know you were just copying tests, but we can get rid of some casting by doing this:

```suggestion
      const result = await getServerSideProps({
        req: { url: '/accountLists/[accountListId]/contacts' },
        query: {
          accountListId: 'account-list-1',
        },
      } as unknown as GetServerSidePropsContext));

      expect(result).toEqual({
        redirect: {
          destination: '/accountLists',
          permanent: false,
        },
      });
```